stellargraph/mapper/sampled_link_generators.py

```diff
@@ -501,6 +503,57 @@ def sample_features(self, head_links, batch_num):
         return batch_feats
 
 
+class Node2VecLinkGenerator(BatchedLinkGenerator):
+    """
+    A data generator for context node prediction with node2vec models.
+
+    At minimum, supply the StellarGraph and the batch size.
+
+    The supplied graph should be a StellarGraph object that is ready for
+    machine learning. Currently the model does not require node features for
+    nodes in the graph.
+
+    Use the :meth:`.flow` method supplying the nodes and targets,
+    or an UnsupervisedSampler instance that generates node samples on demand,
+    to get an object that can be used as a Keras data generator.
+
+    Example::
+
+        G_generator = Node2VecLinkGenerator(G, 50)
+        train_data_gen = G_generator.flow(edge_ids, edge_labels)
+
+    Args:
+        G (StellarGraph): A machine-learning ready graph.
+        batch_size (int): Size of batch of links to return.
+        name (str or None): Name of the generator (optional).
+    """
+
+    def __init__(self, G, batch_size, name=None):
+        super().__init__(G, batch_size, use_node_features=False)
+
+        self.name = name
+
+    def sample_features(self, head_links, batch_num):
+        """
+        Sample the ids of the target and context nodes.
+        and return these as a list of feature arrays for the node2vec algorithm.
+
+        Args:
+            head_links: An iterable of edges to perform sampling for.
+
+        Returns:
+            A list of feaure arrays, with each element being the ids of
+            the sampled target and context node.
+        """
+
+        target_ids = [head_link[0] for head_link in head_links]
+        context_ids = [head_link[1] for head_link in head_links]
+        target_feats = self.graph._get_index_for_nodes(target_ids)
+        context_feats = self.graph._get_index_for_nodes(context_ids)
+        batch_feats = [np.array(target_feats), np.array(context_feats)]
+        return batch_feats
```

Comment:
I think `Graph._get_index_for_nodes` is already supposed to return a numpy array? In which case this whole method could be simplified:
```python
return [self.graph._get_index_for_nodes(ids) for ids in zip(*head_links)]
```

Reply:
In the old version ``Graph._get_index_for_nodes`` only return a list of node ids. It has been upgraded to return a numpy array. Obviously, your script is much more concise.